classes/Currency.php

```diff
@@ -179,14 +179,28 @@ public function __construct($id = null, $idLang = null, $idShop = null)
             if (null === $idLang) {
                 $idLang = Context::getContext()->language->id;
             }
+            $defaultIdLang = null;
+            if ((is_array($this->symbol) && !isset($this->symbol[$idLang]))
```

Comment:
always define the defaultIdLang, remove this condition. You repeat the is_array and isset condition just after. It cost nothing to retrive PS_LANG_DEFAULT

Reply:
alright